### DISCUSSION TITLE
WindowNotificationManager Pop-Ups are no longer working in 11 Preview 5

### DISCUSSION THREAD

**Discussion by DamianSuess** [AvaloniaUI/Avalonia#10236] [Q&A] [answered]
**Describe the bug**
After upgrading to v11 preview 5 the `WindowNotificationManager` (aka: Notification Pop-Ups) are no longer working. In Preview 4, this was consistently showing in the top-left position regardless of the `Position` property's setting. After upgrading to prev5, it no longer shows at all.

Since the change long hasn't been curated yet, I skimmed the [Prev4-to-Prev5 comparison](https://github.com/AvaloniaUI/Avalonia/compare/release/11.0.0-preview4...release/11.0.0-preview5). The only PR I could find was, #9277

I apologize if I'm missing something with a new implementation. If so, could it be added to the [Breaking Changes](https://github.com/AvaloniaUI/Avalonia/wiki/Breaking-Changes) wiki page?

**To Reproduce**

1. Create new `WindowNotificationManager(...)` object setting the TopPage to the main shell window
2. Call using the `.Show(new Notification("title", "message", NotificationType.Information, )` method

Sample code snip from Prism.Avalonia:

* [Initialization of object in MainWindow.xaml.cs](https://github.com/AvaloniaCommunity/Prism.Avalonia/blob/master/samples/SampleMvvmApp/Views/MainWindow.xaml.cs)
* [NotificationService  for calling Show(...)](https://github.com/AvaloniaCommunity/Prism.Avalonia/blob/master/samples/SampleMvvmApp/Services/NotifictionService.cs)

**Expected behavior**
The notification pop-up should appear in the bottom right corner

**Screenshots**
n/a

**Desktop (please complete the following information):**

- OS: Windows 10
- Version 0.11.0-Preview5

**Additional context**
This was uncovered during the Prism.Avalonia upgrade to support v11-Preview5 in [PR #39](https://github.com/AvaloniaCommunity/Prism.Avalonia/pull/39) with the **Samples\SampleMvvmApp** project

**Comment by rabbitism** [comment, marked as answer]
It works fine in ComtrolCatalog demo, the registration is changed. I also changed my code accordingly. Originally in window constructor, now after control template applied. Maybe that's the difference.

**Comment by DamianSuess** [comment]
@rabbitism interesting, I missed that sample. Checking it out now.
Sounds like this one needs to make it onto the breaking changes wiki page for sure.

**Comment by DamianSuess** [comment]
Interesting. So, developers no longer have to define it in the MainWindow, but they can define it in the view's .axaml.cs. In this [example](https://github.com/AvaloniaUI/Avalonia/blob/master/samples/ControlCatalog/Pages/NotificationsPage.xaml.cs), it's being initialized in the `OnAttachedToVisualTree(...)` method.

While reading through the PR again I understand why the change was made. Thank you again @rabbitism

**Comment by timunie** [comment]
Converting this issue into a discussion. If you issue was not solved, please let me know.

**Reply by DamianSuess** [reply]
@timunie, thanks for converting this over. Yes, everything is working out very well. And, thank you for changing it over... one less bug is a good thing 👍 

As per the report's top lines, what sparked off submitting this as a bug was that it was not listed in the [Breaking Changes](https://github.com/AvaloniaUI/Avalonia/wiki/Breaking-Changes) wiki page. Thus, it was unknown what changed. Thankfully @rabbitism pointed me in the right direction.

**Reply by rabbitism** [reply]
@DamianSuess HOHO my pleasure. And hope you feel better now.

**Reply by DamianSuess** [reply]
@rabbitism Doing better each and every day, thank you. Who knew getting old was this much fun 😄
